src/math/mat.zig

```diff
@@ -361,11 +361,24 @@ pub fn Mat(
         }
 
         /// Matrix * Vector multiplication
-        pub inline fn mulVec(a: *const Matrix, b: *const ColVec) ColVec {
+        pub inline fn mulVecR(matrix: *const Matrix, vector: *const ColVec) ColVec {
             var result = [_]ColVec.T{0}**ColVec.n;
             inline for (0..Matrix.rows) |row| {
                 inline for (0..ColVec.n) |i| {
-                    result[i] += a.v[row].v[i] * b.v[row];
+                    result[i] += matrix.v[row].v[i] * vector.v[row];
+                }
+            }
+            return vec.Vec(ColVec.n, ColVec.T){
+                .v = result
+            };
+        }
+
+        /// Vector * Matrix multiplication
+        pub inline fn mulVecL(vector: *const RowVec, matrix: *const Matrix) ColVec{
```

Comment:
I think this is the wrong place to have this code live (which also explains the naming issue you raised):

* `mat.mulVec(vec)` should multiply a `Matrix * Vector`, which you already have implemented
* `vec.mulMat(mat)` should multiply a `Vector * Matrix`, which you are implementing in this PR currently as `mulVecL`.

i.e. this code and the tests could be moved directly into `vec.zig`

Reply:
Ah got you. Moved function to` vec.zig` .